### PR TITLE
#150215879 Fix Migrations

### PR DESCRIPTION
**Pull request by wanjiku-carol**
**What does this PR do ?**

Includes the migrations folder to allow devs to use unified migrations
Adds instructions on how to run migrations, upgrade after changes to the model and resolving database schema conflicts.

**Description of Task to be completed ?**

- Remove `/migrations` from the `.gitignore` file.
- Include migrate and upgrade instructions in README
- Add upgrade instructions after a `git pull -r` to the README
- Add how to resolve Schema Conflicts to README

**How should this be manually tested?**

Run `flask db history` to confirm that there are no conflicts.

**What are the relevant pivotal tracker stories?**

[#150215879](https://www.pivotaltracker.com/story/show/150215879)

### CODE REVIEW

**Comment by andela-milesanmi** [comment]
LGTM

**Comment by LarryWachira** [comment]
@wanjiku-carol Found this on Flask-Migrate:

```
Notes:

All commands also take a --directory DIRECTORY option that points to the directory containing the migration scripts. If this argument is omitted the directory used is migrations.
```

Perhaps you could change the name of the universal migrations folder to eliminate any chances of that folder being modified. And also keep `migrations` ignored so that any local migration folder remain ignored.

**Comment by wanjiku-carol** [comment]
@LarryWachira , Thanks for that. But wouldn't changing the name of the folder and adding the migrations back into the gitignore beat the purpose of the task? When the PR is merged, we will be required to get rid of our local migrations folder and make use of the one migrations folder, then we will no longer have to db init again. But I will refer to @njirap 's infinite wisdom for guidance on this?

**Comment by LarryWachira** [comment]
@wanjiku-carol Good point. However, changing the directory name would not beat the task's purpose in any way. The migrations folder would still exist, we'd just be avoiding any accidental changes to it that could sneak into staging or production. Changing the name would force anyone applying migrations to specify the directory name with each command. But I guess if we and future devs stay keen that requirement isn't entirely necessary.

**Comment by andela-uibrahim** [comment]
@LarryWachira good point you have made so far. however i don't think changing the name is a good idea. lets all work with a unified migration folder like @wanjiku-carol mentioned and push it to github whenever we make changes for others to update.

**Comment by Andela-ddiei** [comment]
Rebase your branch and squash commits.

**Comment by wanjiku-carol** [comment]
@Andela-ddiei Merge conflicts resolved, rebased and squashed.

**Comment by Andela-ddiei** [comment]
@wanjiku-carol You still have conflicts in some files.

**Comment by andela-uibrahim** [comment]
@wanjiku-carol this branch now has 31 file changes 😦 😧 😮 😲 😵 😳 😱 😨 😰. considering the task that were done. this should not be the case. please use git diff to know where the changes are and remove the unnecessary one's

**Comment by wanjiku-carol** [comment]
@andela-uibrahim Every time I rebase and upgrade, new versions are created. So as long as I keep rebasing with changes to the models, files are being changed constantly. I'm currently working on the current Schema conflicts.

**Comment by wanjiku-carol** [comment]
@andela-uibrahim migrations folder updated.

File: api/helpers.py [hunk 1]
```diff
@@ -20,4 +20,4 @@ def save_tags(tags):
             )
             tag.save()
         final_tags.append(tag)
-    return final_tags
+    return final_tags
```
Comment (LarryWachira) [inline api/helpers.py]: No new lines at end of files.

Comment (LarryWachira) [reply api/helpers.py]: Also, squash commits.

**Comment by andela-oowonikoko** [comment]
@wanjiku-carol Please rebase your branch :thank_you:

**Pull request merged**